src/Plugin/Action/UserRoleAdd.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\UserRoleAdd.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\rules\Core\RulesActionBase;
+
+/**
+ * Provides a 'Add user role' action.
+ *
+ * @action(
+ *   id = "rules_user_role_add",
+ *   label = @Translation("Adds roles to a particular user"),
+ *   category = @Translation("User"),
+ *   context = {
+ *     "user" = @ContextDefinition("entity:user",
+ *       label = @Translation("User")
+ *     ),
+ *     "roles" = @ContextDefinition("entity:user_role",
+ *       label = @Translation("Entity"),
+ *       multiple = TRUE
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add port for rules_user_roles_options_list.
+ */
+class UserRoleAdd extends RulesActionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return $this->t('Adds roles to a particular user');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $account = $this->getContextValue('user');
+    $roles = $this->getContextValue('roles');
+    if ($account !== FALSE) {
```

Comment:
That should not be necessary. The user context is required so we don't need to check that. The caller that invokes the action plugin is responsible to make sure that a user object is always there before executing the action.